chrome/content/zotero/xpcom/plugins.js

```diff
@@ -135,7 +136,18 @@ Zotero.Plugins = new function () {
 				Zotero.logError(`Error running bootstrap method '${method}' on ${id}`);
 				Zotero.logError(e);
 			}
-			
+
+			for (let observer of observers) {
+				if (observer[method]) {
+					try {
+						observer[method](params, reason);
```

Comment:
These could actually return promises, so should [return `maybePromise` and `await`](https://github.com/zotero/zotero/blob/9b7b4fe134abd1431ad446c6b3649d1eb3686e91/chrome/content/zotero/xpcom/sync/syncRunner.js#L1325-L1328) so that errors get logged. (I'm actually not handling this above either, but I can fix that separately.)